### DISCUSSION TITLE
How to get current URL or pathname?

### DISCUSSION THREAD

**Discussion by mrvectopus** [shuding/nextra#2773] [Q&A] [answered]
Hello 💟 

I need to get a pathname of current page, for example if I open a page "About" I need get a url directory "/about.en" (without locale `en` it's been perfect 😄 ) and use this in `const data` for db filter.

I found this solution for my page `about.en.mdx`:

```
...
export const getServerSideProps = async ({ params, resolvedUrl }) => {

  const resourceUrl = new URL(`/${resolvedUrl}`);
  console.log(resourceUrl.pathname);

  const data = await db.users.filter("username", "${resourceUrl}").sort

...
```
With this method I'm view correctly pathname in VS console `/about.en` but when I try use this data `${resourceUrl}` in `const data` link (I mean result like this: `.filter("group_slug", "/about.en")`) this is not work...  :(

**Comment by arno-fukuda** [comment, marked as answer]
```
import { useRouter } from "next/router";

const Component = () => {
  const router = useRouter();
  const { pathname, asPath } = router;

  console.log("Pathname:", pathname);
  console.log("As Path:", asPath);

  return <div>hello world</div>;
};

export default Component;

```
